grails-app/services/org/pih/warehouse/product/ProductPackageService.groovy

```diff
@@ -0,0 +1,62 @@
+package org.pih.warehouse.product
+
+import grails.gorm.transactions.Transactional
+import org.pih.warehouse.core.ProductPrice
+
+@Transactional
+class ProductPackageService {
+
+    ProductSupplier save(ProductPackageCommand command) {
+        setPackageData(command)
+        if (command.contractPricePrice) {
+            setContractPriceData(command.productSupplier, command.contractPricePrice, command.contractPriceValidUntil)
+        }
+        return command.productSupplier
+    }
+
+    private static void setPackageData(ProductPackageCommand command) {
```

Comment:
why is this method static (and the one below also)?

Reply:
Because the implementation doesn’t depend on actual instance of the service

Reply:
`setPackageData` sounds like a method that should set something on a currently existing instance, something feels wrong to me with that.

Reply:
it does on the product supplier instance that is bound to the command.